### PR TITLE
Add definitions for built-in functions.

### PR DESCRIPTION
**Pull request by schungx**
@tamasfe   These are definitions for all the built-in functions.

### CODE REVIEW

File: examples/static.d.rhai [hunk 1]
```diff
@@ -7,5 +7,164 @@
...
+/// 
+/// ```rhai
+/// fn foo(x) { }
+/// 
+/// print(is_def_fn("foo", 1));     // prints true
+/// print(is_def_fn("foo", 2));     // prints false
+/// print(is_def_fn("foo", 0));     // prints false
+/// print(is_def_fn("bar", 1));     // prints false
+/// ```
+fn is_def_fn(fn_name: &str, num_params: INT) -> bool;
```
Comment (tamasfe) [inline examples/static.d.rhai:122]: I think we should use `u64` or something like here as the type? If multiple integer types are supported, we should be able to define overloads (not yet implemented though :( ).

File: examples/static.d.rhai [hunk 2]
```diff
@@ -7,5 +7,164 @@
...
+/// If the specified name is not a valid function name, an error is raised.
+/// 
+/// # Example
+/// 
+/// ```rhai
+/// let f = Fn("foo");      // function pointer to 'foo'
+/// 
+/// f.call(42);             // call: foo(42)
+/// ```
+fn Fn(fn_name: &str) -> FnPtr;
```
Comment (tamasfe) [inline examples/static.d.rhai:54]: I don't think `&str` type means too much in Rhai, I believe we should stick to either `string` or `String` for all strings. What do you think?

File: examples/static.d.rhai [hunk 3]
```diff
@@ -7,5 +7,164 @@
...
+/// 
+/// ```rhai
+/// let x = 42;
+/// 
+/// eval("let y = x; x = 123;");
+/// 
+/// print(x);           // prints 123
+/// print(y);           // prints 42
+/// ```
+fn eval(script: &str) -> ?;
```
Comment (tamasfe) [inline examples/static.d.rhai:170]: `&str`, see comments above.

File: examples/static.d.rhai [hunk 4]
```diff
@@ -7,5 +7,164 @@
...
+/// print(is_def_var("foo"));       // prints false
+/// 
+/// {
+///     let y = 1;
+///     print(is_def_var("y"));     // prints true
+/// }
+/// 
+/// print(is_def_var("y"));         // prints false
+/// ```
+fn is_def_var(var_name: &str) -> bool;
```
Comment (tamasfe) [inline examples/static.d.rhai:141]: `&str`, see comments above.

**Review by tamasfe — CHANGES_REQUESTED** [review]

**Comment by tamasfe** [comment]
Thanks! There are some decisions we'll have to make regarding types, I left some comments about those, otherwise LGTM.

File: examples/static.d.rhai (hunk 1 above)
Comment (schungx) [reply examples/static.d.rhai:122]: Well, `INT` can be either `i64` or `i32` depending on a feature flag, so it depends on the Engine.

In most of my docs I just use `i64` (which is the default) and be done with it.  Maybe we should here also.

File: examples/static.d.rhai (hunk 2 above)
Comment (tamasfe) [reply examples/static.d.rhai:54]: Yeah we can do the mapping when we generate the definitions.

File: examples/static.d.rhai (hunk 1 above)
Comment (tamasfe) [reply examples/static.d.rhai:122]: I think I have the syntax for union types (e.g. `type Int = i32 | i64;`) implemented, I can't remember. We can probably use that down the line to deal with this. But yeah it probably shouldn't matter.

Comment (schungx) [reply examples/static.d.rhai:122]: Either way I'm fine.  Let's use `i64` then.  At least then it is consistent with the default set of docs for the standard library.

**Comment by tamasfe** [comment]
I added the missing `?` for the time being since the parser expects a type. I'll have to also change the function return syntax from `:` to `->` as well.

**Pull request merged**